packages/api/src/hooks/useAuthorize.ts

```diff
@@ -1,30 +1,40 @@
-import { useEffect, useMemo } from 'react';
-import { getActiveAuthTokenIDFromLocalStorage } from '@deriv/utils';
+import { useCallback, useMemo } from 'react';
+import { getActiveAuthTokenIDFromLocalStorage, getActiveLoginIDFromLocalStorage } from '@deriv/utils';
 import useFetch from '../useFetch';
+import useInvalidateQuery from '../useInvalidateQuery';
 
 /** A custom hook that authorize the user with the given token. If no token is given,
  * it will use the current token from localStorage.
  */
-const useAuthorize = (token?: string) => {
+const useAuthorize = () => {
     const current_token = getActiveAuthTokenIDFromLocalStorage();
+    const invalidate = useInvalidateQuery();
 
     const { data, ...rest } = useFetch('authorize', {
-        payload: { authorize: token || current_token || '' },
+        payload: { authorize: current_token || '' },
         options: { enabled: Boolean(current_token) },
     });
 
     // Add additional information to the authorize response.
-    const modified_authorize = useMemo(() => ({ ...data?.authorize }), [data?.authorize]);
+    const modified_authorize = useMemo(() => ({ ...data?.authorize }), [data]);
 
-    useEffect(() => {
-        if (current_token !== token) {
-            // Todo: Update the token in the localStorage since we are switching the current account.
-        }
-    }, [current_token, token]);
+    const switchAccount = useCallback(
```

Comment:
@farhan-nurzi-deriv why you are handling the switchaccount in the same file with authorize? I think it's better to give the hooks a single responsibility

Reply:
Because switching account is basically re-authorize with the new token right, so it does the same responsibility 🙇